src/hypertrace/agent/init/__init__.py

```diff
@@ -123,10 +122,13 @@ def init_instrumentation_flask(self, app) -> None:
         '''Creates a flask instrumentation wrapper using the config defined in hypertraceconfig'''
         logger.debug('Calling AgentInit.flaskInit().')
         try:
+            if self.is_registered('flask'):
+                return
             from hypertrace.agent.instrumentation.flask import FlaskInstrumentorWrapper  # pylint: disable=C0415
             self._modules_initialized['flask'] = True
             self._flask_instrumentor_wrapper = FlaskInstrumentorWrapper()
-            self._flask_instrumentor_wrapper.instrument_app(app)
+            if app:
```

Comment:
Should this check happen before we load the instrumentor wrapper?

Reply:
Yes, that check should happen there, but it was missing an important piece. There are two ways to initialize the flask instrumentation module now--this mirrors how the OTel flask instrumentor module works. One calls instrumentation_app, if there is an app object provided. The other acts globally by updating the Flask class definition within the flask module, when no app object is provided.